### ISSUE TITLE
New field log.verbosity

### ISSUE THREAD

**Issue by vbohata** [elastic/ecs#513]
There is a field log.level to capture the log level of the specific event but sometimes we need to know what is the application verbosity setting. I propose to add:
 - log.verbosity

Use case: app setting verbosity is: DEBUG, there will be log events with log.level:ERROR to log.level:DEBUG.

**Comment by vbohata** [comment]
Maybe there should be 2 fields:
 - log.verbosity: DEBUG/INFO ...
 - log.policy: header/payload ...

**Comment by Randy-312** [comment]
I LIke it!
I've been looking to use watcher (or ML) across my many applications to create an incident (SNOW webHook) when some application has a higher than normal number of errors.  While I WAS going to rely upon one of my exiting STDERR / STDOUT fields, your proposal WILL Be better.

Additionally, it's going to allow me to do more around selective retention of my logs.
I don't need to keep the DEBUG messages around nearly as long as the FATAL ones.

**Comment by webmat** [comment]
I understand the usefulness of `log.verbosity`. It can help explain gaps in the logs you're seeing (`log.verbosity:WARN` explains why you're not seeing a `log.level:INFO` message you're looking for). It can also be used to correlate with metrics such as "free disk space" ;-)

I'd like to hear a bit more about `log.policy`. I think I grok it as well. E.g. is Packetbeat configure to log the HTTP payloads as well. Is the idea also to explain gaps in monitoring? Do you see other uses for it?

@felixbarny what do you think of `log.verbosity`?

**Comment by Randy-312** [comment]
I need to revise my earlier enthusiasm, based on a concern.

Since the verbosity level is not passed in the log itself, normally, I'm not sure how this is something that could be added in all cases other than a log event that occurs on startup.

**Comment by webmat** [comment]
Indeed, I don't think that many log sources offer the option of logging the verbosity per se.

However if you're both deploying Beats and configuring your software's verbosity via the same configuration management tool, you can make the change there. The config management tool could set your desired log level at the event source, and add a processor in the relevant Beats config. Perhaps something like this:

```YAML
processors:
- add_fields:
  when:
    equals:
      event.dataset: "myapp.error" # whatever uniquely defines the relevant log
    target: 'log'
    fields:
      verbosity: debug
```
